### DISCUSSION TITLE
how to remove comments label in html source code

### DISCUSSION THREAD

**Discussion by hangyangws** [vercel/next.js#41717] [Help] [answered]
<img width="454" alt="image" src="https://user-images.githubusercontent.com/9067839/197491486-74b03f31-9466-4188-94c7-177bf167e7a5.png">
in my next project,
if the code is  ssr data render.
eg:

`<span className="test">Convert {props.base} to {props.quote}/span>`

and the shource code could like be: 
`<span class="test">Convert<!-- -->base<!-- --> to<!-- -->quote<!-- --></span>`
。

so，how to remove the `<!-- -->`  comments label in html source code?

**Comment by icyJoseph** [comment, marked as answer]
Refer to this answer I gave a while ago: https://github.com/vercel/next.js/discussions/38256#discussioncomment-3070196

So you kind of want to build the line without interpolations:

```jsx
const priceLine = `Convert ${props.base} to ${props.quote}`
```

Or

```jsx
<span className="test">
          {`Convert ${props.base} to ${props.quote}`}
</span>
```

Please do notice that your example has an error, but I've omitted it:

```jsx
                                                                                                             | missing closing <
<span className="test">Convert {props.base} to {props.quote}/span>
```

**Reply by hangyangws** [reply]
thx
